L2SIVacuum/POUs/Functions/Gauges/FB_IMG401.TcPOU

```diff
@@ -0,0 +1,351 @@
+    <Declaration><![CDATA[(* This function is for the Cold Cathode MKS 500. 
+
+If connected to Beckhoff EP boxes. Set the EP bit to TRUE, this is necessary for the 
+IMG401-to-EP box interface because the EP boxes do not natively support the 5v IO 
+signals on the gauge.
+
+This function provides ILK and Set Point Protection for the hot Cathode.
+*)
+
+FUNCTION_BLOCK FB_IMG401 EXTENDS FB_GaugeBase
+VAR_IN_OUT
+		
+END_VAR
+VAR_INPUT
+	PG	:	ST_VG;
+	bEP : BOOL :=FALSE; // Set to True if This Gauge is connected to EP BOX and not EL Terminals
+	tRecoverDelay:TIME:=T#600S; (*Delay Time after the first cycle to start the device. Default is 600S*)
+END_VAR
+VAR_OUTPUT
+{attribute 'pytmc' := '
+    pv: 
+	'}
+	IG : ST_VG;
+END_VAR
+VAR
+	rV : REAL;
+	GaugeTurnOnTmr : TON;
+	tStartupTimer:TON;
+	iTermBits: UINT := 32767 ; // The terminal's maximum value in bits
+	(*IOs to be linked*)
+	/// Controls and I/Os
+	i_iPRESS_R AT %I* :INT; // input Pressure // Link to analog Input
+	q_xHV_DIS AT %Q* : BOOL; // Enable High Voltage when True // 'TcLinkTo' (EP2624) ^Output
+	// only for EL and ES terminal
+	i_xHV_ON AT %I*	: BOOL; //  True when High Voltage is on  // 'TcLinkTo' (EL1084) ^Input
+//	i_xDisc_Active AT %I* : BOOL;// Discharge Current Active // 'TcLinkTo' (EL1124) ^Input
+	binit:BOOL:=TRUE;
+END_VAR
+
+VAR CONSTANT
+	// Ranges has to match the configurator software
+	vBase : REAL := 1.0;
+	vDisconnected : REAL:= 0.0;
+	vTorr : REAL := 10; 
+	vGaugeOff: REAL := 10;
+	//vNoDischarge: REAL := 9.3;
+	MinPressure: REAL := 1E-10;
+	cDefaultPressure : REAL := 0;
+	rDeadband : REAL :=1.0;
+END_VAR
+
+VAR PERSISTENT
+	bWasOn : BOOL;
+	bAutoRecover:BOOL;
+	bAutoRecoverWrite:BOOL;
+END_VAR
+
+
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[(*Startup time*)
+IF (bInit) THEN
+	tStartupTimer.IN:=TRUE;
+	bInit :=FALSE;
+END_IF
+tStartupTimer(PT:=T#3S);
+
+(* 500 Logarithmic Output Conversion, factory default configuration *)
+If (iTermBits=0) THEN iTermBits := 32767;END_IF
+rV := 10*INT_TO_REAL(IG.i_iPRESS_R)/iTermBits;
+
+
+(* 401 Logarithmic Output Conversion, factory default configuration 
+   EL3174, EL3174-xxxx iTermbits 30518, 10V
+*)
+If (iTermBits=0) THEN iTermBits := 30518;END_IF
+rV := 10*INT_TO_REAL(IG.i_iPRESS_R)/iTermBits;
+
+(* Set Guage State based on the Analog voltage*)
+IF rV <= vDisconnected THEN
+	IG.eState := GaugeDisconnected;
+	IG.rPRESS := cDefaultPressure;
+ELSIF rV > vDisconnected AND rV < (vGaugeOff -rDeadband)  THEN
+	IG.eState := Valid;
+	IG.rPRESS := LREAL_TO_REAL(EXPT(10,((rV*vBase)- vTorr)));
+ELSIF rV >= (vGaugeOff -rDeadband) THEN
+	IG.eState := Off;
+	//IG.rPRESS := LREAL_TO_REAL(EXPT(10,((rV-vBase)/vSlope+LOG(pBase))));
+	IG.rPRESS := cDefaultPressure;
+ELSE
+	IG.eState := OoR;
+	//IG.rPRESS := LREAL_TO_REAL(EXPT(10,((rV-vBase)/vSlope+LOG(pBase))));
+	IG.rPRESS := cDefaultPressure;
+END_IF
+
+(* Set Guage State based on the inputs HV ON and Discharge current*)
+If (NOT bEP) THEN
```

Comment:
Does it read the HV status via 5V terminal or 24? because if it has a 24V signal then EP implementation will be the same as the EL. we don't need this if block.

Reply:
Thank you. No need to check EP. You are correct. Modified both IMG401 and CM501 function.